From dfb54792121839b5750480844be9aa27807e0fa5 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Mon, 9 May 2022 09:14:25 +0800
Subject: [PATCH 5/5] drm: drm_file struct kABI compatibility workaround

The recent fix for DRM core
  patches.suse/drm-serialize-drm_file.master-with-a-new-spinlock.patch
introduced a new field in struct drm_file.
Apply the standard workaround to move it at the tail with __GENKSYMS__
for keeping the kABI compatibility; drm_file isn't embedded, so it
should be safe to do that.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 include/drm/drm_file.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/drm/drm_file.h b/include/drm/drm_file.h
index 2f6420a35b5a..628f44cf1d4d 100644
--- a/include/drm/drm_file.h
+++ b/include/drm/drm_file.h
@@ -213,9 +213,6 @@ struct drm_file {
 	 */
 	struct drm_master *master;
 
-	/** @master_lock: Serializes @master. */
-	spinlock_t master_lookup_lock;
-
 	/** @pid: Process that opened this file. */
 	struct pid *pid;
 
@@ -329,6 +326,10 @@ struct drm_file {
 
 	/* private: */
 	unsigned long lock_count; /* DRI1 legacy lock count */
+#ifndef __GENKSYMS__
+	/** @master_lock: Serializes @master. */
+	spinlock_t master_lookup_lock;
+#endif
 };
 
 /**
-- 
2.25.1

